pop3/commands/pass/pass.go

```diff
@@ -10,11 +14,40 @@ import (
 // may only be given in the AUTHORIZATION state immediately
 // after a successful USER command
 func Process(c *Client, cmd Command) Reply {
-	res := OK
-	msg := ""
 	if c.State != Authorization { //TODO USER must be last command issued by client as well
-		res = ERR
-		msg = "invalid state"
+		return Reply{Result: ERR, Message: "invalid state"}
 	}
-	return Reply{Result: res, Message: msg}
+	if c.GetBool("authenticated") {
+		return Reply{Result: ERR, Message: "already authenticated"}
+	}
+	lastData := strings.Split(c.GetString("last_command"), ":")
+	if len(lastData) != 2 {
+		return Reply{Result: ERR, Message: "invalid arguments"}
+	}
+	lastCMD, user := lastData[0], lastData[1]
+	if lastCMD != "USER" {
+		return Reply{Result: ERR, Message: "illegal user and pass sequence"}
+	}
+	c.Set("last_command", "PASS")
```

Comment:
pass can only be followed by a successful user, i just smashed the last command they send with the user they want to log in as 1 value in the map and got them out later. i could have used separate keys but that felt just as dirty